internal/convenience/decoder/decoder.go

```diff
@@ -73,7 +74,23 @@ func (o *OutputDecoder) HandleOutputV2(
 		"inputIndex", processOutputData.InputIndex,
 		"outputIndex", processOutputData.OutputIndex,
 	)
-	if processOutputData.Payload[2:10] == model.VOUCHER_SELECTOR {
+
+	input := &model.InputEdge{
+		Node: struct {
+			Index int    `json:"index"`
+			Blob  string `json:"blob"`
+		}{
+			Blob: processOutputData.Payload,
+		},
+	}
+	convertedInput, err := o.GetConvertedInput(*input)
```

Comment:
Aqui você pode fazer igual ao RetrieveDestination logo abaixo. Não precisaria converter e passar para outro método.

Reply:
I did it this way because I only need the AppContract, which is one of the attributes returned in the response object of this method (GetConvertedInput), to populate the objects ConvenienceNotice and ConvenienceVoucher